definitions/npm/idb_v5.x.x/flow_v0.83.x-/idb_v5.x.x.js

```diff
@@ -1,427 +1,643 @@
 // Derived from TS typings at https://github.com/jakearchibald/idb/blob/c8bab898f01bacab96097f87c1e42a0c19e01425/lib/idb.d.ts
+// Updated for changes between v2 and v5.
 
-type IDBArrayKey = Array<IDBValidKey>;
-type IDBValidKey = number | string | Date | IDBArrayKey;
+declare type $idb$IDBArrayKey = $ReadOnlyArray<$idb$IDBValidKey>;
```

Comment:
All types are defined outside a module definition so that they can be reused. Indeed this library has 2 very similar entries, that differ only by a few methods.

Reply:
Is there any concern of "pollution" from having these outside the module definition? Might be a silly question, I haven't worked with Flow in a little while...

Reply:
Yeah, that will pollute the global scope, that's why I prefixed the names with some prefix.
This is pretty common in the libdefs because we were _always_ polluting the global scope when using commonjs style (with `declare module.exports: XXX`). Only with the module style (`declare export XXX`) the global scope isn't polluted anymore.

See for example https://github.com/flow-typed/flow-typed/blob/master/definitions/npm/algoliasearch_v3.x.x/flow_v0.71.x-v0.103.x/algoliasearch_v3.x.x.js.

I'd be happy to make the prefix more "difficult" to collide with if you think that's needed!

Reply:
And I thought it was better (and especially more future-proof) to pollute the global scope than copy-pasting everything.